pkg/virt-api/webhooks/validating-webhook/admitters/vms-admitter.go

```diff
@@ -238,62 +231,78 @@ func ValidateVirtualMachineSpec(field *k8sfield.Path, spec *v1.VirtualMachineSpe
 	return causes
 }
 
-func validateNoModificationsDuringRename(ar *v1beta1.AdmissionRequest, vm *v1.VirtualMachine) []metav1.StatusCause {
-	var causes []metav1.StatusCause
-
-	hasRenameReq := hasRenameRequest(vm)
+func validateStateChangeRequests(ar *v1beta1.AdmissionRequest, vm *v1.VirtualMachine) []metav1.StatusCause {
+	// Only rename request is validated
+	renameRequest := getRenameRequest(vm)
 
 	// Prevent creation of VM with rename request
 	if ar.Operation == v1beta1.Create {
-		if hasRenameReq {
-			return append(causes, metav1.StatusCause{
+		if renameRequest != nil {
+			return []metav1.StatusCause{{
 				Type:    metav1.CauseTypeFieldValueInvalid,
 				Message: "Creating a VM with a rename request is not allowed",
 				Field:   k8sfield.NewPath("Status", "stateChangeRequests").String(),
-			})
+			}}
 		}
 	} else if ar.Operation == v1beta1.Update {
 		existingVM := &v1.VirtualMachine{}
 		err := json.Unmarshal(ar.OldObject.Raw, existingVM)
 
 		if err != nil {
-			return append(causes, metav1.StatusCause{
+			return []metav1.StatusCause{{
 				Type:    metav1.CauseTypeUnexpectedServerResponse,
 				Message: "Could not fetch old VM",
-			})
+			}}
 		}
 
-		existingVMHasRenameReq := hasRenameRequest(existingVM)
-
-		if existingVMHasRenameReq {
-			return append(causes, metav1.StatusCause{
+		if getRenameRequest(existingVM) != nil {
+			return []metav1.StatusCause{{
 				Type:    metav1.CauseTypeFieldValueInvalid,
 				Message: "Modifying a VM during a rename process is not allowed",
-			})
+			}}
 		}
+	}
 
-		// Reject rename requests if the VM is running
-		if hasRenameReq {
-			runningStatus, _ := vm.RunStrategy()
-			if runningStatus != v1.RunStrategyHalted {
-				return append(causes, metav1.StatusCause{
-					Type:    metav1.CauseTypeFieldValueInvalid,
-					Message: "Cannot rename a running VM",
-					Field:   k8sfield.NewPath("spec", "running").String(),
-				})
-			}
-		}
+	if renameRequest == nil {
+		return nil
 	}
 
-	return causes
+	// Reject rename requests if the VM is running
+	runningStatus, _ := vm.RunStrategy()
+	if runningStatus != v1.RunStrategyHalted {
+		return []metav1.StatusCause{{
+			Type:    metav1.CauseTypeFieldValueInvalid,
+			Message: "Cannot rename a running VM",
+			Field:   k8sfield.NewPath("spec", "running").String(),
+		}}
+	}
+
+	newName, hasNewName := renameRequest.Data["newName"]
+	if !hasNewName || newName == "" {
```

Comment:
checking that `newName` is empty here is wrong.
Since a user can set an empty string as a new name and the error in result should be that this is not a legal name not that the name was not provided.

"" is not a DNS subdomain since DNS subdomain start and end with an alphanumeric character.

Reply:
Done